consensus/obcpbft/pbft-core.go

```diff
@@ -439,25 +440,52 @@ func (instance *pbftCore) committed(digest string, v uint64, n uint64) bool {
 // =============================================================================
 
 // handle new consensus requests
+func (instance *pbftCore) requestSync(msgPayload []byte, senderID uint64) error {
+	msg := &Message{&Message_Request{&Request{Payload: msgPayload,
+		ReplicaId: senderID}}}
+	instance.manager.inject(pbftMessageEvent{
```

Comment:
The way I envision these state machine objects working, is an event comes in, is processed, and if it needs to, it emits a new event.

So, for instance, the batch timer expires, it builds the batch, and then would return/emit a `pbftMessageEvent`, this gets received, decoded, and it turns out it is a request, so it returns a `pbftRequestEvent`, this gets processed, and, if primary, returns a `pbftPrePrepareEvent`, which returns a `pbftPrepareEvent`, and finally, the chain is done.

This way it's much easier to inject specific events into the state machine, and the entry path would be the same for all types.  As the code stands today, it's very odd that we inject requests through so many different method calls, ultimately, I see all internal injection through `processEvent` and external injection carefully curated via calls from the consensus API.